keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
 
-    rpc ProvisionerGrantBucketAccess (ProvisionerGrantBucketAccessRequest) returns (ProvisionerGrantBucketAccessResponse);
-    rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
+When a user deletes a BucketClaim, the corresponding secret/serviceaccount are also deleted. If a pod has that secret mounted when delete is called, then the secret will not be deleted, but will instead have its deletionTimestamp set. In this way, access to a Bucket is preserved until the application pod dies. 
+
+When an admin deletes any of the Class objects, it does not affect existing Buckets as they store a copy of the fields in the class object. If a Bucket  is manually deleted by an admin, then the cluster is considered to be in an invalid state. Manual recovery is possible if data is not already lost.  
+
+# COSI API Reference
+
+## Bucket
+
+Resource to represent a Bucket in OSP. Buckets are cluster-scoped.
+
+```yaml
+Bucket {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketSpec {
+    // Provisioner is the name of driver associated with this bucket
+    Provisioner string
+
+    // DeletionPolicy is used to specify how COSI should handle deletion of this
+    // bucket. There are 3 possible values:
+    //  - Retain: Indicates that the bucket should not be deleted from the OSP
+    //  - Delete: Indicates that the bucket should be deleted from the OSP
+    //        once all the workloads accessing this bucket are done
+    //  - ForceDelete: Indicates that the bucket and its contents should be
+    //        deleted from the OSP even if workloads are currently accessing
+    //        this bucket
+    DeletionPolicy DeletionPolicy
+
+    // Name of the BucketClass specified in the BucketRequest
+    // +optional
+    BucketClassName  string
+
+    // Name of the BucketClaim that resulted in the creation of this Bucket
+    // +optional
+    BucketClaim corev1.ObjectReference
+
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
+
+    // Parameters is an opaque map for passing in configuration to a driver
+    // for creating the bucket
+    // +optional
+    Parameters map[string]string
+
+    // BucketID is the unique id of the bucket in the OSP. This field should be
+    // used to specify a bucket that has been created outside of COSI.
+    //
+    // This field will be empty when the Bucket is dynamically provisioned by COSI.
+    // +optional
+    BucketID string
+  }
+
+  Status BucketStatus {
+    // BucketReady is a boolean condition to reflect the successful creation
+    // of a bucket.
+    BucketReady bool
+  }
```

Comment:
An earlier draft defined `status.bucketID` as the name of the backend bucket. In this version of the kep I don't see where an admin can see the name of the backend bucket.

Reply:
BucketID can be found here: https://github.com/kubernetes/enhancements/pull/2813/files/6cb304f0453fc621c27793f75a536b26e094d298#diff-fc5fdb05b4b40c902c57b03ae47bd25c5061f1b9f710bfd057e2332c42c2fa7eR574#L574, just a few lines above this comment